core/base/batch_utilities.hpp

```diff
@@ -121,14 +144,58 @@ auto unbatch(const InputType* batch_object)
 }
 
 
-template <typename ValueType, typename IndexType, typename OutputType>
+namespace detail {
+
+
+template <typename ValueType, typename IndexType>
+void assert_same_sparsity_in_batched_data(
+    const std::vector<gko::matrix_data<ValueType, IndexType>>& data)
+{
+    auto num_nnz = data.at(0).nonzeros.size();
```

Comment:
TBH, I would have prefered
```suggestion
    if(data.empty()){
      return;
    }
    auto num_nnz = data.at(0).nonzeros.size();
```
I think it should be valid to check empty data.